src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Reflection/SignatureTypes.cs

```diff
@@ -312,6 +355,8 @@ public static void MakeSignatureConstructedGenericType(Type genericTypeDefinitio
                     Assert.False(et.IsGenericTypeParameter);
                     Assert.True(et.IsGenericMethodParameter);
                     Assert.Equal(5, et.GenericParameterPosition);
+                    Assert.Throws<NotSupportedException>(() => et.IsValueType);
```

Comment:
Move up before `Type et = t.GenericTypeArguments[0];` line and change to Assert.False.

Reply:
In this case, I wanted to make sure the new code that was added for IsValueType\IsEnum throws when the generic parameter is a signature type.

If I compare `t` with `t.IsValueType` that is just testing normal reflection and will return `true` for `Span<sigtype>` and `false` for the `List<sigtype>`.

Reply:
That area now checks both `t.IsValueType` and `t.IsEnum` against the same property on `genericTypeDefinition`.